tests/test_vxlan_global.rb

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.expand_path('../ciscotest', __FILE__)
+require File.expand_path('../../lib/cisco_node_utils/vxlan_global', __FILE__)
+
+include Cisco
+
+# TestVxlanGlobal - Minitest for VxlanGlobal node utility
+class TestVxlanGlobal < CiscoTestCase
+  def setup
+    super
+    no_feature_vxlan_global
+  end
+
+  def teardown
+    no_feature_vxlan_global
+    super
+  end
+
+  def no_feature_vxlan_global
+    config('no feature fabric forwarding')
+  end
+
+  def test_feature_on_off
+    feat = VxlanGlobal.new
+    feat.feature_enable
+    assert(VxlanGlobal.feature_enabled)
+
+    feat.feature_disable
+    refute(VxlanGlobal.feature_enabled)
+  end
+
+  def test_dup_host_ip_addr_detection_set
+    vxlan_global = VxlanGlobal.new
+    val = [200, 20]
+    vxlan_global.feature_enable
+    vxlan_global.dup_host_ip_addr_detection_set(' ', val[0], val[1])
+    assert_equal(val, vxlan_global.dup_host_ip_addr_detection,
+                 'Error: fabric forwarding dup_host_ip_addr_detection ' \
+                 'get values mismatch')
```

Comment:
Note: These error messages are fine to include but they're optional. I've found that in simple test cases like this they don't add a lot of value since on failure the assert is going to display an error that already tells you the feature area (the test filename) ; the api being tested (the method name) ; and the expected values versus actual results: <[200,20]> expected but was <foo>. By all means keep these here if you want, just know that they're only required when the assert failure needs clarity.